lms/views/basic_lti_launch.py

```diff
@@ -368,4 +369,9 @@ def configure_assignment(self):
 
         self.context.js_config.maybe_enable_grading()
 
+        # Re-do enable_lti_launch_mode to account for any changes after the
+        # document has been selected
+        JSConfig._hypothesis_client.fget.cache_clear()
+        self.context.js_config.enable_lti_launch_mode()
```

Comment:
This is now on master https://github.com/hypothesis/lms/pull/3465

If I'm not missing anything I think you can completly drop https://github.com/hypothesis/lms/pull/3458/commits/3f83afaed8936f696e8bceb08b25c7c78b90f031 from this branch.